### PR TITLE
feat(bindings/crypto-js): Implement `OlmMachine.sign`

### LINKED ISSUE #797: bindings/crypto-nodejs: Expose function to sign arbitrary objects
This was available off the OlmMachine in the old bindings: https://github.com/matrix-org/matrix-rust-sdk-bindings/blob/main/crates/matrix-sdk-crypto-nodejs/src/machine.rs#L375

### PR DESCRIPTION
**Pull request by Hywan**
This PR adds the following API and types:

- [x] `OlmMachine.crossSigningStatus` along with its `CrossSigningStatus` type,
- [x] `DeviceKeyId`, `DeviceKeyAlgorithm` and `DeviceKeyAlgorithName` types,
- [x] `OlmMachine.sign`, along with `Ed25519Signature`, `Signatures`, `Signature`, `MaybeSignature`.

This patch fixes #797.

### CODE REVIEW

**Review by turt2live — APPROVED** [review]

File: bindings/matrix-sdk-crypto-nodejs/src/identifiers.rs [hunk 1]
```diff
@@ -94,6 +95,109 @@ impl DeviceId {
...
+    #[napi(getter)]
+    pub fn device_id(&self) -> DeviceId {
+        self.inner.device_id().to_owned().into()
+    }
+
+    /// Return the device key ID as a string.
+    #[napi]
+    #[allow(clippy::inherent_to_string)]
+    pub fn to_string(&self) -> String {
+        self.inner.as_str().to_owned()
```
Comment (jplatte) [inline bindings/matrix-sdk-crypto-nodejs/src/identifiers.rs:137]:
```suggestion
        self.inner.to_string()
```

Comment (jplatte) [reply bindings/matrix-sdk-crypto-nodejs/src/identifiers.rs:137]: (seems simpler)

File: bindings/matrix-sdk-crypto-nodejs/src/identifiers.rs [hunk 2]
```diff
@@ -94,6 +95,109 @@ impl DeviceId {
...
+    #[napi(getter)]
+    pub fn name(&self) -> DeviceKeyAlgorithmName {
+        self.inner.clone().into()
+    }
+
+    /// Return the device key algorithm as a string.
+    #[napi]
+    #[allow(clippy::inherent_to_string)]
+    pub fn to_string(&self) -> String {
+        self.inner.as_ref().to_owned()
```
Comment (jplatte) [inline bindings/matrix-sdk-crypto-nodejs/src/identifiers.rs:167]:
```suggestion
        self.inner.to_string()
```

File: bindings/matrix-sdk-crypto-nodejs/src/types.rs [hunk 3]
```diff
@@ -0,0 +1,157 @@
...
+#[napi]
+impl MaybeSignature {
+    /// Check whether the signature has been successfully decoded.
+    #[napi(getter)]
+    pub fn is_valid(&self) -> bool {
+        self.inner.is_ok()
+    }
+
+    /// Check whether the signature could not have been successfully
+    /// decoded.
```
Comment (jplatte) [inline bindings/matrix-sdk-crypto-nodejs/src/types.rs:136]:
```suggestion
    /// Check whether the signature couldn't be successfully
    /// decoded.
```

File: bindings/matrix-sdk-crypto-nodejs/src/types.rs [hunk 4]
```diff
@@ -0,0 +1,157 @@
...
+    }
+
+    /// The signature, if successfully decoded.
+    #[napi(getter)]
+    pub fn signature(&self) -> Option<Signature> {
+        self.inner.as_ref().cloned().map(Into::into).ok()
+    }
+
+    /// The base64 encoded string that is claimed to contain a
+    /// signature but could not be decoded if any.
```
Comment (jplatte) [inline bindings/matrix-sdk-crypto-nodejs/src/types.rs:149]:
```suggestion
    /// signature but could not be decoded, if any.
```

File: bindings/matrix-sdk-crypto-nodejs/src/identifiers.rs (hunk 1 above)
Comment (Hywan) [reply bindings/matrix-sdk-crypto-nodejs/src/identifiers.rs:137]: :+1:

File: bindings/matrix-sdk-crypto-nodejs/src/identifiers.rs (hunk 2 above)
Comment (Hywan) [reply bindings/matrix-sdk-crypto-nodejs/src/identifiers.rs:167]: :+1:

File: bindings/matrix-sdk-crypto-nodejs/src/types.rs (hunk 3 above)
Comment (Hywan) [reply bindings/matrix-sdk-crypto-nodejs/src/types.rs:136]: :+1:

File: bindings/matrix-sdk-crypto-nodejs/src/types.rs (hunk 4 above)
Comment (Hywan) [reply bindings/matrix-sdk-crypto-nodejs/src/types.rs:149]: :+1:

**Pull request merged**